src/config.rs

```diff
@@ -223,6 +224,112 @@ impl Config {
         Ok(())
     }
 
+    /// update the config.terms map from the contents of a file
+    /// in case any 'term:' header is used within the file
+    pub async fn update_terms_from_file(
+        &mut self,
+        doc_id: &str,
+        data: &str,
+    ) -> ftd::p1::Result<()> {
+        /// returns header key and value
+        /// given header string
+        fn segregate_key_value(
+            header: &str,
+            doc_id: &str,
+            line_number: usize,
+        ) -> ftd::p1::Result<(String, String)> {
+            if !header.contains(':') {
+                return Err(ftd::p1::Error::ParseError {
+                    message: format!(": is missing in: {}", header),
+                    doc_id: doc_id.to_string(),
+                    line_number,
+                });
+            }
+
+            let mut parts = header.splitn(2, ':');
+            match (parts.next(), parts.next()) {
+                (Some(name), Some(value)) => {
+                    // some header and some non-empty value
+                    Ok((name.trim().to_string(), value.trim().to_string()))
+                }
+                (Some(name), None) => Err(ftd::p1::Error::ParseError {
+                    message: format!("Unknown header value for header \'{}\'", name),
+                    doc_id: doc_id.to_string(),
+                    line_number,
+                }),
+                _ => Err(ftd::p1::Error::ParseError {
+                    message: format!("Unknown header found \'{}\'", header),
+                    doc_id: doc_id.to_string(),
+                    line_number,
+                }),
+            }
+        }
+
+        /// converts the given string to document_id
+        /// and returns it
+        fn convert_to_document_id(doc_id: &str) -> String {
```

Comment:
Why is this a private function? Looks like we have to do this in many places, do we not already have a function? Also the argument name itself is `doc_id`, which is short for `document_id`, so it converts to `doc_id`, why is it taking `doc_id` as input? 

Give some input/output samples. Also read https://fpm.dev/glossary/ to ensure you are using the terminology correctly.